app/controllers/reliefs/AvoidanceSchemesController.scala

```diff
@@ -16,57 +16,54 @@
 
 package controllers.reliefs
 
-import config.FrontendDelegationConnector
-import connectors.BackLinkCacheConnector
-import controllers.{AtedBaseController, BackLinkController}
-import controllers.auth.{AtedFrontendAuthHelpers, AtedRegime}
 import connectors.{BackLinkCacheConnector, DataCacheConnector}
-import controllers.{AtedBaseController, BackLinkController}
-import controllers.auth.{AtedFrontendAuthHelpers, AtedRegime, ClientHelper}
+import controllers.BackLinkController
+import controllers.auth.{AuthAction, ClientHelper}
 import forms.ReliefForms._
 import models.TaxAvoidance
-import services.ReliefsService
-import uk.gov.hmrc.play.frontend.auth.DelegationAwareActions
-import play.api.i18n.Messages.Implicits._
 import play.api.Play.current
+import play.api.i18n.Messages.Implicits._
+import play.api.mvc.{Action, AnyContent}
+import services.{DelegationService, ReliefsService}
 
 trait AvoidanceSchemesController extends BackLinkController
-  with AtedFrontendAuthHelpers with ReliefHelpers with DelegationAwareActions  with ClientHelper {
+  with AuthAction with ReliefHelpers with ClientHelper {
 
-  def view(periodKey: Int) = AuthAction(AtedRegime) {
-    implicit atedContext =>
+  def view(periodKey: Int): Action[AnyContent] = Action.async { implicit request =>
+    authorisedAction { implicit authContext =>
       validatePeriodKey(periodKey) {
-        reliefsService.retrieveDraftReliefs(atedContext.user.atedReferenceNumber, periodKey).flatMap {
-          case Some(x) if (x.reliefs.isAvoidanceScheme.contains(true))  =>
+        reliefsService.retrieveDraftReliefs(authContext.atedReferenceNumber.get, periodKey).flatMap {
```

Comment:
RED - None.get risk.

Reply:
Removed and refactored to use String as opposed to Option String